simulator/risc_v/riscv_instr.h

```diff
@@ -0,0 +1,116 @@
+/* riscv_instr.h - instruction parser for risc_v
+ * @author Alexandr Misevich
+ * Copyright 2018 MIPT-MIPS
+ */
+
+#ifndef RISCV_INSTR_H
+#define RISCV_INSTR_H
+
+// Generic C++
+#include <cassert>
+#include <array>
+#include <unordered_map>
+
+// MIPT-MIPS modules
+#include <infra/types.h>
+#include <infra/macro.h>
+#include <infra/string/cow_string.h>
+
+#include "riscv_register/riscv_register.h"
+
+class RISCVInstr
+{
+    private:
+        uint32 instr = NO_VAL32;
+
+        RISCVRegister src1 = RISCVRegister::zero;
+        RISCVRegister src2 = RISCVRegister::zero;
+        RISCVRegister dst = RISCVRegister::zero;
+        
+        uint32 v_src1 = NO_VAL32;
+        uint32 v_src2 = NO_VAL32;
+        uint64 v_dst = NO_VAL64;
```

Comment:
RISCV has no HI/LO, so it is also RegisterUInt32

Reply:
My bad, I meant RegisterUInt

Reply:
So, again template?